kindelia_core/src/hvm.rs

```diff
@@ -1681,6 +1682,52 @@ impl Runtime {
     return Ok(done);
   }
 
+  // Full-reduce a term.
+  // TODO: rewrite as stack based? not needed?
+  pub fn normalize(&mut self, host: Loc, mana:u64, seen: &mut HashSet<RawCell>) -> Result<RawCell, RuntimeError> {
+    let term  = ask_lnk(self, host);
+    if seen.contains(&term) {
+      Ok(term)
+    }
+    else {
+      let term = reduce(self, host, mana)?;
+      seen.insert(term);
+      let mut recursive_locs = vec![];
+      match term.get_tag() {
+        CellTag::DP0 => {
+          recursive_locs.push(term.get_loc(2));
+        },
+        CellTag::DP1 => {
+          recursive_locs.push(term.get_loc(2));
+        },
+        CellTag::LAM => {
+          recursive_locs.push(term.get_loc(1));
+        },
+        CellTag::APP => {
+          recursive_locs.push(term.get_loc(0));
```

Comment:
I don't understand why you put `term.get_loc(0)` two times in `CellTag::App` and `CellTag::Sup`

Reply:
OH! Yeah, it should have been a 0 and a 1 in both cases.